tests/slurm_command_gen_strategy/test_chakra_replay.py

```diff
@@ -75,19 +64,23 @@ def test_parse_slurm_args(
         expected_result: Dict[str, Any],
         slurm_system: SlurmSystem,
     ) -> None:
-        slurm_args = self.get_slurm_args(slurm_system, job_name_prefix, env_vars, cmd_args, num_nodes, nodes)
+        slurm_args = cmd_gen_strategy._parse_slurm_args(job_name_prefix, env_vars, cmd_args, num_nodes, nodes)
         assert slurm_args["image_path"] == expected_result["image_path"]
         assert slurm_args["container_mounts"] == expected_result["container_mounts"]
 
-    def test_parse_slurm_args_invalid_cmd_args(self, slurm_system: SlurmSystem) -> None:
+    def test_parse_slurm_args_invalid_cmd_args(
+        self, cmd_gen_strategy: ChakraReplaySlurmCommandGenStrategy, slurm_system: SlurmSystem
+    ) -> None:
         job_name_prefix = "chakra_replay"
         env_vars = {"NCCL_DEBUG": "INFO"}
         cmd_args = {"trace_path": "/workspace/traces/"}  # Missing "docker_image_url"
         num_nodes = 2
         nodes = ["node1", "node2"]
 
-        with pytest.raises(KeyError):
-            self.get_slurm_args(slurm_system, job_name_prefix, env_vars, cmd_args, num_nodes, nodes)
+        with pytest.raises(KeyError) as exc_info:
+            cmd_gen_strategy._parse_slurm_args(job_name_prefix, env_vars, cmd_args, num_nodes, nodes)
+
+        assert str(exc_info.value) == "'docker_image_url'", "Expected missing docker_image_url key"
```

Comment:
Such issues should be handled on TestDefinition level. Let's leave it as is for now and will remove this code later.

Reply:
Would you like me to remove any SlurmCommandGenStrategy tests that raise exceptions, which should be handled by TestDefinition? I will create a follow-up PR if that’s the case.